### PR TITLE
Fixes for best binding pullaway bug.

### PR DESCRIPTION
**Pull request by ssepeq**
- Repointed the pullaway undo to the new gp vector;
- Added feature to continually rotate ligand during dock to maintain best-bind pairs;
- Added option for aliphatic best-binding moieties with a size limit;
- Added ftn to conform best binding residues to ligand before iterations;
- Limited eligibility of aromatic side chains for best binding if CA-CB angle is too far;
- Fixed a bug with best-binding ligand atoms being reused;
- Update forces_by_Z to include bindings.dat entries where one atom is *;
- H-bond potential is now proportional to h-bonding strength;
- Proactively decreased potential bonding for hydrophilic+hydrophobic combinations.
- Added MOV_FORBIDDEN constant for ruling out motion if any forbidding bit is set;
- Added information about best-binding selection to .dock file output;
- Prioritized best-binding pairs in pullaway undo;
- No longer append new movie to old movie;
- Had to update the OR1A1 test to use predicted pi-stacking residues;
- Removed errant comma in viewer.

### CODE REVIEW

**Comment by ssepeq** [comment]
🟩🟩🟩

**Comment by ssepeq** [comment]
🟩🟩🟩

File: src/primarydock.cpp [hunk 1]
```diff
@@ -429,21 +497,22 @@ void iteration_callback(int iter)
         for (l=0; l<3; l++)
         {
             #if _use_gloms
-            if (ligand_gloms[l].atoms.size() && sc_gloms[l].aminos.size())
+            if (gp.size() > l)
```
Comment (primaryodors) [inline src/primarydock.cpp:500]: Let's refactor gp to whatever it stands for.

**Review by primaryodors — CHANGES_REQUESTED** [review]
Just one item that should be a quick fix in VSCode.

**Review by primaryodors — APPROVED** [review]

**Pull request merged**